modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/jaxrs/extension/RelationshipsResourceImpl.java

```diff
@@ -0,0 +1,169 @@
+ * details.
+ */
+
+package com.liferay.object.rest.internal.jaxrs.extension;
+
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectRelationship;
+import com.liferay.object.rest.dto.v1_0.ObjectEntry;
+import com.liferay.object.rest.extension.RelationshipsResource;
+import com.liferay.object.rest.manager.v1_0.ObjectEntryManager;
+import com.liferay.object.rest.manager.v1_0.ObjectEntryManagerTracker;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectRelationshipLocalService;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.Parameters;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
+
+import java.net.URI;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import javax.validation.constraints.NotNull;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.UriInfo;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+@Component(
+	factory = "com.liferay.object.rest.extension.RelationshipsResource",
+	property = {"api.version=v1.0", "osgi.jaxrs.resource=true"},
+	service = RelationshipsResource.class
+)
+@Path("/v1.0")
+public class RelationshipsResourceImpl implements RelationshipsResource {
+
+	/**
+	 * Invoke this method with the command line:
+	 *
+	 *  curl http://localhost:8080/o/headless-admin-user/{currentObjectEntryId}/{objectRelationshipName} -u test@liferay.com:test
+	 */
+	@GET
+	@Override
+	@Parameters(
+		{
+			@Parameter(in = ParameterIn.PATH, name = "currentObjectEntryId"),
+			@Parameter(in = ParameterIn.PATH, name = "objectRelationshipName"),
+			@Parameter(in = ParameterIn.QUERY, name = "page"),
+			@Parameter(in = ParameterIn.QUERY, name = "pageSize")
+		}
+	)
+	@Path("/{currentObjectEntryId}/{objectRelationshipName}")
+	@Produces({"application/json", "application/xml"})
+	@Tags({@Tag(name = "ObjectEntry")})
+	public Page<ObjectEntry> getCurrentObjectEntriesObjectRelationshipNamePage(
+			@NotNull @Parameter(hidden = true)
+			@PathParam("currentObjectEntryId")
+			Long currentObjectEntryId,
+			@NotNull @Parameter(hidden = true)
+			@PathParam("objectRelationshipName")
+			String objectRelationshipName,
+			@Context Pagination pagination)
+		throws Exception {
+
+		List<ObjectRelationship> objectRelationships =
```

Comment:
Maybe it is a good idea to extract all the code that we want to move to the service in the future to a private method to separate the logics and better communicate with Sierra team what we need on the services (or that would be easier for us to do it in the future